### PR TITLE
Stack recipe CLI improvements

### PR DESCRIPTION
**Pull request by wjayesh**
## Describe changes
Some improvements to the CLI:
- fix "string indices must be integers" bug
- A warning is displayed on pull that overwriting will delete all tfstate files.
- The code now checks for kubectl, helm and docker installation in addition to terraform.
- There's now an option to skip checking `locals.tf` file when you run the `deploy` function.
- `zenml` version in the generated YAML file now automatically reflects the current `zenml` version.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by htahir1** [comment]
@wjayesh why have you not tagged people here so they know to review? :-)

File: src/zenml/cli/stack_recipes.py [hunk 1]
```diff
@@ -81,28 +81,97 @@ def __init__(self, path: str) -> None:
...
+            raise RuntimeError(
+                "helm is not installed on your machine or not available on  "
+                "your $PATH. It is required for stack recipes to create releases "
+                "on Kubernetes. Please visit "
+                "https://helm.sh/docs/intro/install/ "
+                "to install it."
+            )
+        if not self._is_docker_installed():
+            raise RuntimeError(
+                "docker is not installed on your machine or not available on  "
```
Comment (htahir1) [inline src/zenml/cli/stack_recipes.py:116]:
```suggestion
                "Docker is not installed on your machine or not available on  "
```

File: src/zenml/cli/stack_recipes.py [hunk 2]
```diff
@@ -81,28 +81,97 @@ def __init__(self, path: str) -> None:
...
+                "kubectl is not installed on your machine or not available on  "
+                "your $PATH. It is used by stack recipes to create some "
+                "resources on Kubernetes and to configure access to your "
+                "cluster. Please visit "
+                "https://kubernetes.io/docs/tasks/tools/#kubectl "
+                "to install it."
+            )
+        if not self._is_helm_installed():
+            raise RuntimeError(
+                "helm is not installed on your machine or not available on  "
```
Comment (htahir1) [inline src/zenml/cli/stack_recipes.py:108]:
```suggestion
                "Helm is not installed on your machine or not available on  "
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Nice, small changes only

File: src/zenml/cli/stack_recipes.py [hunk 3]
```diff
@@ -81,28 +81,97 @@ def __init__(self, path: str) -> None:
         self.tf = python_terraform.Terraform(working_dir=str(path))
 
     def check_installation(self) -> None:
-        """Checks if terraform is installed on the host system.
+        """Checks if necessary tools are installed on the host system.
 
         Raises:
-            RuntimeError: if terraform is not installed.
+            RuntimeError: if any tool is not installed.
```
Comment (strickvl) [inline src/zenml/cli/stack_recipes.py:87]:
```suggestion
            RuntimeError: if any required tool is not installed.
```

File: src/zenml/cli/stack_recipes.py [hunk 4]
```diff
@@ -820,14 +905,15 @@ def deploy(
             git_stack_recipes_handler: The GitStackRecipesHandler instance.
             stack_recipe_name: The name of the stack_recipe.
             path: The path at which you want to install the stack_recipe(s).
-            force: Force the run of the stack_recipe.
+            force: Force pull the stack recipe overwriting any existing files.
```
Comment (strickvl) [inline src/zenml/cli/stack_recipes.py:908]:
```suggestion
            force: Force pull the stack recipe, overwriting any existing files.
```

**Review by strickvl — APPROVED** [review]
Just a couple of minor things, but otherwise looks good to me.

File: src/zenml/cli/stack_recipes.py [hunk 5]
```diff
@@ -81,28 +81,97 @@ def __init__(self, path: str) -> None:
...
-        ret_code, _, _ = self.tf.cmd("-version")
-        return bool(ret_code == 0)
+        try:
+            self.tf.cmd("-version")
+        except FileNotFoundError:
+            return False
+
+        return True
+
+    def _is_kubectl_installed(self) -> bool:
```
Comment (schustmi) [inline src/zenml/cli/stack_recipes.py:137]: I think your solution works perfectly fine, I'll just suggest an alternative here that might save some code: You can use `shutil.which("command")` to check if a command exists without actually running it.
```python
def command_available(command: str) -> bool:
    return shutil.which(command) is not None
    
def check_installation(self):
    if not self.command_available("kubectl"):
        raise ...
```

**Review by schustmi — APPROVED** [review]
Looks good! Just an optional suggestion to remove a few lines of code, but feel free to ignore. The linting issues and installation issues on windows 3.7/3.8 should be gone once you merge the current develop branch into yours

**Pull request merged**